cashctrl_ledger/tests/test_accounts.py

```diff
@@ -125,44 +121,64 @@ def test_modify_account_with_invalid_group_raise_error(self, engine):
+        expected_categories = ACCOUNTS["group"].to_list()
+        assert not set(expected_categories).issubset(initial_categories), (
+            "Expected categories already exists"
+        )
 
-        assert not initial_accounts[initial_accounts["group"].str.startswith("/Balance")].empty, (
-            "There are no remote accounts placed in /Balance node"
+        # Ensure target categories not present on remote are created when mirroring
+        engine.accounts.mirror(ACCOUNTS, delete=True)
+        accounts = engine.accounts.list()
+        categories = engine._client.list_categories("account", include_system=True)
+        categories = categories["path"].to_list()
+        assert not accounts[accounts["group"].str.startswith("/Balance")].empty, (
+            "Accounts with '/Balance' root category were not created"
+        )
+        assert set(expected_categories).issubset(categories), (
+            "Expected categories were not created"
         )
 
-        engine.accounts.mirror(expected.copy(), delete=True)
-        mirrored_df = engine.accounts.list()
-        updated_categories = engine._client.list_categories("account", include_system=True)
-        updated_categories_dict = updated_categories.set_index("path")["number"].to_dict()
-        difference = set(categories_dict.keys()) - set(updated_categories_dict.keys())
-        initial_sub_nodes = [
-            key for key in difference if key.startswith("/Balance") and key != "/Balance"
-        ]
-
-        assert mirrored_df[mirrored_df["group"].str.startswith("/Balance")].empty, (
-            "Accounts placed in /Balance node were not deleted"
+        # Ensure orphaned categories except root nodes are deleted when mirroring
+        engine.accounts.mirror(initial_accounts, delete=True)
+        accounts = engine.accounts.list()
+        categories = engine._client.list_categories("account", include_system=True)
+        categories = categories["path"].to_list()
+        assert_frame_equal(initial_accounts, accounts)
+        assert set(initial_categories) == set(categories), (
+            "Some orphaned categories were not deleted"
         )
-        assert len(initial_sub_nodes) > 0, "Sub-nodes were not deleted"
-        assert updated_categories_dict["/Balance"] == categories_dict["/Balance"], (
-            "Root node /Balance was deleted"
+        assert "/Balance" in set(categories), (
+            "Mirroring initial accounts should not delete root '/Balance' category"
         )
 
-        engine.accounts.mirror(initial_accounts.copy(), delete=True)
-        mirrored_df = engine.accounts.list()
-        updated_categories = engine._client.list_categories("account", include_system=True)
-        updated_categories_dict = initial_categories.set_index("path")["number"].to_dict()
-        pd.testing.assert_frame_equal(initial_accounts, mirrored_df)
-        assert updated_categories_dict == categories_dict, "Some categories were not restored"
```

Comment:
We should also check if mirroring an empty account chart leaves the four root categories and deletes all other categories. If this is too complicated or time consuming, we should at least test test that erasing all accounts in a given root category erases all sub-nodes, but leaves the root category.